bee-network/bee-autopeering/src/local/mod.rs

```diff
@@ -0,0 +1,241 @@
+    pub(crate) fn public_key(&self) -> PublicKey {
+        *self.read().public_key()
+    }
+
+    /// Returns the current private salt of this identity.
+    pub(crate) fn private_salt(&self) -> Option<Salt> {
+        self.read().private_salt().cloned()
+    }
+
+    /// Sets a new private salt.
+    pub(crate) fn set_private_salt(&self, salt: Salt) {
+        self.write().set_private_salt(salt);
+    }
+
+    /// Returns the current public salt of this identity.
+    pub(crate) fn public_salt(&self) -> Option<Salt> {
+        self.read().public_salt().cloned()
+    }
+
+    /// Sets a new public salt.
+    pub(crate) fn set_public_salt(&self, salt: Salt) {
+        self.write().set_public_salt(salt);
+    }
+
+    /// Signs a message using the private key.
+    pub(crate) fn sign(&self, msg: &[u8]) -> Signature {
+        self.read().sign(msg)
+    }
+
+    /// Adds a service to this local peer.
+    pub fn add_service(&self, service_name: impl ToString, protocol: ServiceProtocol, port: u16) {
+        self.write().add_service(service_name, protocol, port);
+    }
+
+    /// Returns the list of services this identity supports.
+    pub(crate) fn services(&self) -> ServiceMap {
+        self.read().services().clone()
+    }
+
+    fn read(&self) -> RwLockReadGuard<LocalInner> {
+        self.inner.read().expect("error getting read access")
+    }
+
+    fn write(&self) -> RwLockWriteGuard<LocalInner> {
+        self.inner.write().expect("error getting write access")
+    }
+}
+
+impl LocalInner {
+    fn peer_id(&self) -> &PeerId {
+        &self.peer_id
+    }
+
+    fn public_key(&self) -> &PublicKey {
+        self.peer_id().public_key()
+    }
+
+    fn private_salt(&self) -> Option<&Salt> {
+        self.private_salt.as_ref()
+    }
+
+    fn set_private_salt(&mut self, salt: Salt) {
+        self.private_salt.replace(salt);
+    }
+
+    fn public_salt(&self) -> Option<&Salt> {
+        self.public_salt.as_ref()
+    }
+
+    fn set_public_salt(&mut self, salt: Salt) {
+        self.public_salt.replace(salt);
+    }
+
+    fn sign(&self, msg: &[u8]) -> Signature {
+        self.private_key.sign(msg)
+    }
+
+    fn add_service(&mut self, service_name: impl ToString, protocol: ServiceProtocol, port: u16) {
+        self.services.insert(service_name, protocol, port)
+    }
+
+    fn services(&self) -> &ServiceMap {
+        &self.services
+    }
+}
+
+impl fmt::Debug for Local {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Local")
+            .field("peer_id", &self.read().peer_id())
+            .finish()
+    }
+}
+
+impl fmt::Display for Local {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.read().peer_id())
+    }
+}
+
+impl Eq for Local {}
+impl PartialEq for Local {
+    fn eq(&self, other: &Self) -> bool {
+        self.read().peer_id() == other.read().peer_id()
+    }
+}
+
+impl Default for LocalInner {
```

Comment:
Weren't we supposed to remove this ?

Reply:
That was for `PeerId`. But for `Local` it also makes sense.

Reply:
Removed, and replaced with a `Local::generate` function.